cli/src/bin/subg/main.rs

```diff
@@ -47,22 +47,20 @@ pub(crate) enum PoolFormat {
 
 impl Display for PoolFormat {
     fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
-        let s = format!("{:?}", self);
+        let s = format!("{self:?}");
         write!(f, "{}", s.to_lowercase())
     }
 }
 
 fn parse_pool_path(pool_path: &str) -> (&Path, PoolFormat) {
     let path = Path::new(pool_path);
-    let format = match path.extension() {
-        Some(ext) => match ext.to_str().unwrap() {
-            "json" => PoolFormat::Json,
-            "yaml" | "yml" => PoolFormat::Yaml,
-            _ => {
-                eprintln!("Unknown pool file extension: {}", ext.to_str().unwrap());
-                exit(exitcode::USAGE);
-            }
-        },
+    let format = match path.extension().map(|v| v.to_string_lossy()).as_deref() {
```

Comment:
Although I imagine it is very rare that an extension would not be utf-8, as part of this update, I eliminated the `unwrap`s

Reply:
Why the lossy version and not `to_str`?

Reply:
`to_str` may fail 
![image](https://github.com/milton-hirsch-institute/subnet-garden/assets/4666989/73e4f920-d554-49ee-b4ec-b9778680ecfe)

Reply:
So what happens if someone feeds in a unicode string? Can you write a test for that?

Reply:
I went to write such a test, but then I noticed that since the path is created from a str, the extension must be str-compatible. Was that what you were trying to guide me to?

I pushed a change using `to_str` instead